### PR TITLE
[CCR] Update es monitoring mapping and

### PR DESCRIPTION
**Pull request by martijnvg**
change qa tests to query based on leader index.

### CODE REVIEW

File: x-pack/plugin/core/src/main/resources/monitoring-es.json [hunk 1]
```diff
@@ -916,6 +916,83 @@
...
             }
           }
+        },
+        "ccr_stats": {
+          "properties": {
+            "leader_index": {
+              "type": "keyword"
+            },
+            "shard_id": {
+              "type": "keyword"
```
Comment (jasontedor) [inline x-pack/plugin/core/src/main/resources/monitoring-es.json]: I am not sure about this being a keyword, I think probably a numeric field type is preferred here?

**Review by jasontedor — COMMENTED** [review]
I left one comment.

As we continue to iterate on the stats endpoint (we will) we will need to consider a way to enforce that we keep the mapping up to date with the changes?

File: x-pack/plugin/core/src/main/resources/monitoring-es.json (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/core/src/main/resources/monitoring-es.json]: Good point. It is serialized as a string now (`[index-name][shard-id]`). I think that is not super useful and storing just the shard id is better? (the leader index is already stored in another field)

**Comment by martijnvg** [comment]
> As we continue to iterate on the stats endpoint (we will) we will need to consider a way to enforce that we keep the mapping up to date with the changes?

Agreed. I'm not sure how though. Add unit test in monitoring that iterates over all static parse fields, parse the mapping and then verify if there is a entry in the mapping?

File: x-pack/plugin/core/src/main/resources/monitoring-es.json (hunk 1 above)
Comment (jasontedor) [reply x-pack/plugin/core/src/main/resources/monitoring-es.json]: I think I am missing something, it should only be being serialized as the numerical shard ID, without the index name. See this test:

https://github.com/elastic/elasticsearch/blob/23f12e42c13878b0ff3b4c33420eb1570ee586cd/x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/ccr/CcrStatsMonitoringDocTests.java#L142

and the x-content logic:

https://github.com/elastic/elasticsearch/blob/23f12e42c13878b0ff3b4c33420eb1570ee586cd/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/ShardFollowNodeTaskStatus.java#L380

Comment (martijnvg) [reply x-pack/plugin/core/src/main/resources/monitoring-es.json]: whoops. I read `shardId` as field name and I assumed `ShardId` class.

**Comment by ycombinator** [comment]
Maybe its too heavy handed but Monitoring has [integration tests](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/integration/MonitoringIT.java). Maybe one of those to collect stats and then query for them, to ensure that all the fields were indeed indexed as expected?

**Comment by martijnvg** [comment]
@ycombinator Thanks, I did not know that. I think we can add a test here or somewhere else that uses `CcrStatsCollector` directly, get a monitoring doc based on a dummy ccr stat response, index that monitor doc via an Exporter and then expect that we can search on all field that are in `ShardFollowNodeTaskStatus`.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**